src/include/storage/storage_structure/lists/list_sync_state.h

```diff
@@ -43,27 +47,29 @@ class ListSyncState {
     inline uint32_t getEndElemOffset() const { return startElemOffset + numValuesToRead; }
     inline bool hasValidRangeToRead() const { return UINT32_MAX != startElemOffset; }
     inline uint32_t getNumValuesToRead() const { return numValuesToRead; }
-    inline uint64_t getNumValuesInList() const { return numValuesInList; }
     inline ListSourceStore getListSourceStore() const { return sourceStore; }
-    inline void setSourceStore(ListSourceStore sourceStore) { this->sourceStore = sourceStore; }
-    inline void setDataToReadFromUpdateStore(bool dataToReadFromUpdateStore_) {
-        dataToReadFromUpdateStore = dataToReadFromUpdateStore_;
-    }
-    inline bool hasDataToReadFromUpdateStore() const { return dataToReadFromUpdateStore; }
     inline list_header_t getListHeader() const { return listHeader; }
-    inline void setListHeader(list_header_t listHeader_) { listHeader = listHeader_; }
+    inline uint32_t getNumValuesInList() {
+        return sourceStore == ListSourceStore::PERSISTENT_STORE ? numValuesInPersistentStore :
+                                                                  numValuesInUpdateStore;
+    }
 
-    bool hasMoreToRead();
+    bool hasMoreAndUpdateIfNecessary();
```

Comment:
i would suggest rename this function to `hasMoreToReadAndUpdateRangesIfNecessary`

Reply:
Renamed to `hasMoreAndSwitchSourceIfNecessary`